### ISSUE TITLE
Cannot use @index class decorator when installed from npm

### ISSUE THREAD

**Issue by Doom100500** [szokodiakos/typegoose#201]
The latest npm package was published before 4 mounts, but documentation contains reference to @index.

Any workaround will help much

**Comment by Ben305** [comment]
The problem is that the NPM token has expired and I have no access to the NPM project. That‘s why new releases are not published at the moment.

**Comment by Doom100500** [comment]
Thanks for response.
As workaround i just copied the typescript files files locally to my project...

**Comment by Ben305** [comment]
I'm trying to publish to NPM again asap.

**Comment by zacronos** [comment]
As a workaround for now, you can use `npm install --save "typegoose@https://github.com/StimulatingSolutions/typegoose.git"`.  It's not ideal, but it will get the job done for now -- I suggest anyone who chooses to use this workaround hit the `Subscribe` button for this issue so you can be notified when things are fixed and you can go back to using the npm release.

The reason for the forked version is that right now, `https://github.com/szokodiakos/typegoose` includes only the source files, not the transpiled js files, so `npm install --save "typegoose@https://github.com/szokodiakos/typegoose.git"` doesn't do the job.  @Ben305, if you want to change that so that folks can use `typegoose@https://github.com/szokodiakos/typegoose.git` instead of my fork, you could temporarily remove `lib` from the `.gitignore` file, build, and then commit the `lib` directory to github.  (Or, if it is easier for you, I can PR those changes from my fork.)

**Comment by zacronos** [comment]
Ignore my earlier post (deleted), something wasn't working as I thought.

The odd thing is that version 5.4.0 *seems* to be published to npm, but in terms of functionality it's not 5.4.0.

**Comment by zacronos** [comment]
To clarify my comment, it seems that someone (maybe @Ben305 ?) did manage to publish a 5.4.0 version to npm about a week ago, but the wrong code was published -- when you install via npm, the code still does not contain the `@index` feature, even though the package.json file says it is version 5.4.0.

**Comment by Ben305** [comment]
I‘ll check that later.

**Comment by MateuszKicinski** [comment]
I'm not sure if this is the same issue but it's related, after installing version 5.4.1 and setting a text index on some properties I get these warnings:

`(node:14111) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.`
`(node:14111) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead.`
`(node:14115) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.`
`(node:14115) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead.`
`(node:14115) DeprecationWarning: collection.remove is deprecated. Use deleteOne, deleteMany, or bulkWrite instead.`
`(node:14111) DeprecationWarning: collection.update is deprecated. Use updateOne, updateMany, or bulkWrite instead.`
`(node:14111) DeprecationWarning: collection.findAndModify is deprecated. Use findOneAndUpdate, findOneAndReplace or findOneAndDelete instead.`

**Comment by zacronos** [comment]
@MateuszKicinski  Those warnings are not related.  Try using:
```javascript
mongoose.set('useCreateIndex', true);
mongoose.set('useFindAndModify', false);
mongoose.connect(MONGODB_URI, {
  useNewUrlParser: true,
});
```

That should handle most of the warnings you listed.  If some remain, they are likely because of your code, i.e. you are using `SomeModel.remove()` instead of `SomeModel.deleteOne()` or `SomeModel.deleteMany()`, etc.

**Comment by Doom100500** [comment]
I confirm that after upgrade to the latest version from npm the index decorator can be used. So this issue can be closed i think...

**Comment by zacronos** [comment]
I can also confirm that `@index()` works with the v5.4.1 release on npm.  (@Doom100500, I think you can close the issue since you created it.)

**Comment by Ben305** [comment]
Fixed in 5.4.1
